src/databricks/labs/ucx/assessment/assessment.py

```diff
@@ -60,20 +87,229 @@ def _verify_ws_client(w: WorkspaceClient):
             msg = "Current user is not a workspace admin"
             raise RuntimeError(msg)
 
-    def table_inventory(self, executor: CommandExecutor | None = None):
-        logger.info("Started dataset inventorization...")
-        ce = self._get_command_executor(executor, language=Language.SCALA)
-        params = {"SCHEMA": self._inventory_schema}
-        cmd_code = self._get_command("create_table_inventory.scala", params=params)
-        command_output = ce.run(cmd_code)
-        logger.debug(command_output)
-        if executor is None:
-            self._remove_command_executor()
-        logger.info("Completed dataset inventorization...")
-
-    def compile_report(self):
-        logger.info("Started report compilation...")
-        ce = self._get_command_executor(None, language=Language.SCALA)
-        self.table_inventory(ce)
-        self._remove_command_executor()
-        logger.info("Completed report compilation...")
+    def generate_ext_loc_list(self):
+        crawler = ExternalLocationCrawler(self._backend, ws, self._inventory_catalog,
+                                          self._inventory_schema)
+        return crawler.snapshot()
+
+    def generate_job_assessment(self):
+        crawler = JobsCrawler(self._backend, ws, "hive_metastore",
+                              self._inventory_catalog)
+        return crawler.snapshot()
+
+    def generate_cluster_assessment(self):
+        crawler = ClustersCrawler(self._backend, ws, "hive_metastore",
+                                  self._inventory_catalog)
+        return crawler.snapshot()
+
+
+    # @staticmethod
+    # def _backend(ws: WorkspaceClient, warehouse_id: str | None = None) -> SqlBackend:
+    #     if warehouse_id is None:
+    #         return RuntimeBackend()
+    #     return StatementExecutionBackend(ws, warehouse_id)
+
+
+class InventoryTableCrawler(CrawlerBase):
+
+    def __init__(self, sbe: SqlBackend, catalog, schema):
+        super().__init__(sbe, catalog, schema, "tables")
+
+    def snapshot(self) -> list[Table]:
+        return self._snapshot(self._try_fetch, self._mock_loader)
+
+    def _mock_loader(self):
+        return None
+
+    def _try_fetch(self) -> list[Table]:
+        for row in self._fetch(
+            f'SELECT * FROM {self._schema}.{self._table}'
+        ):
+            yield Table(*row)
+
+
+class ExternalLocationCrawler(CrawlerBase):
+
+    def __init__(self, sbe: SqlBackend, ws: WorkspaceClient, catalog, schema):
```

Comment:
```suggestion
    def __init__(self, backend: SqlBackend, inventory_schema):
```
we always persist this to hive_metastore, no need to specify it. also workspace client is not used anywhere.

Reply:
Done

Reply:
WS is require for the Mounts crawler